### DISCUSSION TITLE
OIDC Support?

### DISCUSSION THREAD

**Discussion by ckwalsh** [shlinkio/discussions#58] [Ideas]
Has OIDC support ever been considered as an alternative to cli generated api keys? I feel like it would make Shlink much easier to integrate into existing systems.

**Comment by ckwalsh** [comment]
I've got a prototype working, that generates api keys on the fly from OIDC tokens, and seems to be working.

It needs a some cleanup, tests, integration with CLI tooling, and documentation, I plan on submitting a merge request after getting the first two in order, but probably won't be able to get back to it until next weekend.

I split the changes across 3 commits:
- Add a "source" column to the api_keys table, to differentiate "normal" keys and keys from OIDC
- Add a ApiKeyBuilder class and hook it in. This is similar to ApiKeyMeta, but is a little easier to work with for my implementation in the last diff
- Add OidcService and hook it into AuthenticationMiddleware

It's been a while since I've touched modern open source PHP, and I'm not super familiar with the factory system in the Shlink code, but I think I figured it out close enough.

**Reply by acelaya** [reply]
Please, don't spend more time on this. It would add a lot of complexity to Shlink and it's not something I want to support.
